src/cmdstan/command.hpp

```diff
@@ -594,6 +594,149 @@ int command(int argc, const char *argv[]) {
     return_code = stan::services::standalone_generate(
         model, fitted_params.samples.block(0, meta_cols, num_rows, num_cols),
         random_seed, interrupt, logger, sample_writers[0]);
+  } else if (user_method->arg("log_prob")) {
+    string_argument *upars_file = dynamic_cast<string_argument *>(
+        parser.arg("method")->arg("log_prob")->arg("unconstrained_params"));
+    string_argument *cpars_file = dynamic_cast<string_argument *>(
+        parser.arg("method")->arg("log_prob")->arg("constrained_params"));
+    bool jacobian_adjust
+        = dynamic_cast<bool_argument *>(
+              parser.arg("method")->arg("log_prob")->arg("jacobian_adjust"))
+              ->value();
+    if (upars_file->is_default() && cpars_file->is_default()) {
+      msg << "No input parameters provided, cannot calculate log-probability";
+      throw std::invalid_argument(msg.str());
+    }
+
+    /*
+      By only updating the sizes/vectors if there is data present, the
+      calculation and printing can be agnostic from the combinations of
+      un-/constrained parameters passed
+    */
+    size_t u_params_vec_size = 0;
+    size_t u_params_size = 0;
+    size_t c_params_vec_size = 0;
+    size_t c_params_size = 0;
+
+    std::vector<double> u_params_r;
```

Comment:
declare these as low as possible so it's easier to line up where they're defined and where they're set;  for example, `c_params` should move lower to above where `c_params are sorted out.  I'd remove space in between declaration and conditional defining so the code proximity makes it clear what's defining them